### PR TITLE
Prepare for the 0.14.0 release

### PR DESCRIPTION
**Pull request by jdneo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,15 @@ All notable changes to the "vscode-java-test" extension will be documented in th
...
-## 0.14.0
+## 0.14.0 - [Unreleased]
 
 ## Added
 - Add the log level setting. [#555](https://github.com/Microsoft/vscode-java-test/issues/555)
+- Navigate to the source code from the test explorer. [#558](https://github.com/Microsoft/vscode-java-test/issues/558)
+
+## Changed
+- Redesign the user experience of running tests with configurations. [#524](https://github.com/Microsoft/vscode-java-test/issues/524)
+  > More details can be found [here](https://aka.ms/java-test-config)
```
Comment (Eskibear) [inline CHANGELOG.md]: Just to confirm if it's your intention to use the block quotes `>` here? I think the mark is often used to quote something, might not be the best choice here. My personal preference is as following:

#### alternative a
- Redesign ....
  - More details .....

#### alternative b
- Redesign .....   [More details](#)

File: README.md [hunk 2]
```diff
@@ -36,6 +36,8 @@ You can find the following commands in the Command Palette (<kbd>**F1**</kbd> /
 |---|---|---|
 | `java.test.report.position` | Specify where to show the test report. Supported values are: `sideView`, `currentView`. | `sideView` |
 | `java.test.log.level` | Specify the level of the test logs. Supported values are: `error`, `info`, `verbose`. | `info` |
+| `java.test.config` | Specify the configuration for the test cases to run with.<br> More details can be found [here](https://aka.ms/java-test-config). | `{}` |
```
Comment (Eskibear) [inline README.md]: Specify the configuration for the test cases to run with.<br>More details can be found [here](https://aka.ms/java-test-config)

vs

Specify the configuration for the test cases to run with. [More details](#)

vs

Specify the configuration for the test cases to run with. <br>[More details](#)

File: src/extension.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ async function doActivate(_operationId: string, context: ExtensionContext): Prom
         testReportProvider,
         logger,
         languages.registerCodeLensProvider({ scheme: 'file', language: 'java' }, testCodeLensProvider),
-        instrumentAndRegisterCommand(JavaTestRunnerCommands.OPEN_DOCUMENT, async (node: TestTreeNode) => await openTextDocument(Uri.file(node.fsPath), node.range)),
+        instrumentAndRegisterCommand(JavaTestRunnerCommands.OPEN_DOCUMENT, async (uri: Uri, range?: Range) => await openTextDocument(uri, range)),
```
[the remark this answers is not in the document]
Comment (jdneo) [reply src/extension.ts]: This is to fix a regression in #582

File: README.md (hunk 2 above)
Comment (jdneo) [reply README.md]: Use the second option

File: CHANGELOG.md (hunk 1 above)
Comment (jdneo) [reply CHANGELOG.md]: Use the alternative b

**Pull request merged**